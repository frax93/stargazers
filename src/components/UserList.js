import { View, StyleSheet, Text, FlatList, Image } from 'react-native';

export default function UserList({ data }) {
  return (
    <View>
      {data ? data?.length > 0 ? <FlatList
        testID="user-list"
        data={data}
        keyExtractor={(item, index) => index}
        renderItem={({ item }) => (
          <View style={styles.itemRow}>
            <Image testID="image-item" source={{ uri: item?.avatar_url || '' }} style={styles.imageViewContainer} />
            <View style={styles.textViewContainer}>
              <Text style={styles.itemTitle} testID="avatar-item">
                {item?.login || ''}
              </Text>
            </View>
          </View>
        )}
      /> : <Text testID="no-element">No element found</Text> : null}
    </View>
  );
}

const styles = StyleSheet.create({
  itemRow: {
    direction: 'rtl',
    flex: 1,
    flexDirection: 'row-reverse',
    alignItems: 'center'
  },
  imageViewContainer: {
    width: 44,
    height: 44,
    margin: 10,
    borderRadius: 22,
  },
  textViewContainer: {
    width: 'auto',
    paddingTop: 10,
  },
  itemTitle: {
    width: '100%',
    textAlign: 'right',
  },
});
