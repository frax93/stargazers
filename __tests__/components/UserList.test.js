import UserList from 'components/UserList';
import { render } from '@testing-library/react-native';

const setup = (props = {}) => {
  const wrapper = render(<UserList {...props} />);
  return wrapper;
};

describe('Testing user list component', () => {
  it("renders without errors", () => {
    const userListComponent = setup();

    const componentToJSON = userListComponent.toJSON();

    expect(componentToJSON).toMatchSnapshot();
  });

  it("renders no element found", () => {
    const userListComponent = setup({
      data: []
    });

    const noTextElement = userListComponent.getByTestId('no-element');

    expect(noTextElement.children[0]).toBe('No element found');
  });

  it("renders an element found", () => {
    const data = [{
      avatar_url: 'avatar',
      login: 'username',
    }];

    const userListComponent = setup({
      data
    });

    const imageItems = userListComponent.getAllByTestId('image-item');
    const avatarItems = userListComponent.getAllByTestId('avatar-item');

    const [firstImageItem] = imageItems || [];
    const [firstAvatarItem] = avatarItems || [];

    expect(firstImageItem.props.source).toHaveProperty('uri', 'avatar');

    expect(firstAvatarItem.children[0]).toBe('username');
  });
});

