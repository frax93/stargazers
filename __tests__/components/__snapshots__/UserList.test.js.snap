// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing user list component renders without errors 1`] = `<View />`;
